prior/__init__.py

```diff
@@ -134,65 +137,44 @@ def _get_git_lfs_cmd():
         return git_lfs_path
 
 
-def load_dataset(
-    dataset: str,
-    revision: str = "main",
-    entity: str = "allenai",
-    offline: bool = False,
-    **kwargs: Any,
-) -> DatasetDict:
-    """Load the dataset from the given revision.
-
-    Args:
-        dataset: The name of the dataset to load.
-        revision: The git revision of the dataset to load. Can be specified as either
-            a commit id sha, tag, or branch. If None, the latest commit to main
-            will be used.
-        entity: The github organization or username that has the dataset.
-        offline: If True, don't attempt to download the dataset from github.
-        kwargs: Allows you to specify variants of a particular dataset (e.g., do you
-            want the variant with a locobot or a different agent?).
-
-    Returns:
-        A DatasetDict containing the loaded dataset.
-    """
-
+def _clone_repo(
+    base_dir: str, entity: str, project: str, revision: str, offline: bool
+) -> Tuple[str, str]:
     def get_cached_sha() -> Optional[str]:
-        if os.path.exists(f"{dataset_dir}/cache"):
-            with LockEx(f"{dataset_dir}/cache-lock"):
-                with open(f"{dataset_dir}/cache", "r") as f:
+        if os.path.exists(f"{project_dir}/cache"):
+            with LockEx(f"{project_dir}/cache-lock"):
+                with open(f"{project_dir}/cache", "r") as f:
                     cache = json.load(f)
                 if revision in cache:
                     return cache[revision]
         return None
 
-    dataset_dir = f"{DATASET_DIR}/{entity}/{dataset}"
-    os.makedirs(dataset_dir, exist_ok=True)
-    start_dir = os.getcwd()
+    project_dir = f"{base_dir}/{entity}/{project}"
```

Comment:
Can we make `project_dir` a parameter passed to `get_cached_sha` or define it above `get_cached_sha`?

Reply:
done